src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -0,0 +1,374 @@
+        return self.get_param('max_concurrent_sessions')
+
+    def set_argument_max_concurrent_sessions(self, max_concurrent_sessions):
+        self.set_param('max_concurrent_sessions', max_concurrent_sessions)
+
+    def get_argument_ready_session_instances(self):
+        return self.get_param('ready_session_instances')
+
+    def set_argument_ready_session_instances(self, ready_session_instances):
+        self.set_param('ready_session_instances', ready_session_instances)
+
+    def get_argument_image(self):
+        return self.get_param('image')
+
+    def get_argument_container_name(self):
+        return self.get_param('container_name')
+
+    def get_argument_cpu(self):
+        return self.get_param('cpu')
+
+    def get_argument_memory(self):
+        return self.get_param('memory')
+
+    def get_argument_env_vars(self):
+        return self.get_param('env_vars')
+
+    def get_argument_startup_command(self):
+        return self.get_param('startup_command')
+
+    def get_argument_args(self):
+        return self.get_param('args')
+
+    def get_argument_target_port(self):
+        return self.get_param('target_port')
+
+    def get_argument_registry_server(self):
+        return self.get_param("registry_server")
+
+    def get_argument_registry_pass(self):
+        return self.get_param("registry_pass")
+
+    def get_argument_registry_user(self):
+        return self.get_param("registry_user")
+
+    def get_environment_client(self):
+        return ManagedEnvironmentClient
+
+    def show(self):
+        try:
+            return self.client.show(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                name=self.get_argument_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def list(self):
+        try:
+            return self.client.list(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def delete(self):
+        try:
+            return self.client.delete(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                name=self.get_argument_name(),
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+
+class SessionPoolCreateDecorator(SessionPoolPreviewDecorator):
+    def validate_arguments(self):
+        validate_container_app_name(self.get_argument_name(), AppType.SessionPool.name)
+        container_type = self.get_argument_container_type()
+        environment_name = self.get_argument_managed_env()
+
+        location = self.get_argument_location()
+        # location = validate_environment_location(self.cmd, location)
+        _ensure_location_allowed(self.cmd, location, CONTAINER_APPS_RP, "sessionPools")
+        self.set_argument_location(location)
```

Comment:
The location in line 163 used to `self.set_argument_location(location)` is meaningless.